### ISSUE TITLE
Online-docs: dbtonovdoc issues

### ISSUE THREAD

**Issue by fsundermeyer** [openSUSE/daps#337]
1. ~~contents of abstract elements is missing in DB4~~ (fixed, see 532b233)
2. ~~space before option element inside screen is missing~~ (fixed, see 6d14d7d)
3. ~~noprofile stylesheet for DB4 generates empty textobjects~~ (fixed, see cb3a27e)
4. ~~db4->novdoc conversion only works when using db5to4.xsl for the db5->db4 conversion first. When using db5to4-withdoctype.xsl for the db5->db4 conversion, it produces invalid NovDoc XML~~ (see https://github.com/openSUSE/daps/issues/337#issuecomment-164384110)
5. ~~&lt;entry&gt; elements in a table that contain a remark cause errors:~~
   - ~~the `remark`-element is deleted~~
   - ~~the `para` element is wrapped with another `para`: &lt;para&gt;&lt;para&gt;...&lt;/para&gt;&lt;/para&gt;~~ (see https://github.com/openSUSE/daps/issues/337#issuecomment-164370187)
6. ~~some screens are garbled (see screenshot)~~ (fixed, see 6d14d7d)
   ![db5_to_novdoc_error](https://cloud.githubusercontent.com/assets/1875958/11749584/f74ef720-a02e-11e5-9078-67739568a491.png)
7. ~~Seem the space between &productname; &productnumber;  is missing~~ (fixed, see 9736b6b)
8. ~~The booktitle in xrefs is resolved to "&lt;space>TITLE&lt;space>"~~ (fixed, see  f8b92f6)
9. ~~subtitle element is missing, e.g. for the Geo Quick Start:~~ (fixed, see 39b2e83)
10. ~~abstract in appendix is missing (eg. `ha_troubleshooting.xml`)~~ (fixed, see d554f47)

**Comment by fsundermeyer** [comment]
How to reproduce "Files converted with db5to4-withdoctype.xsl have a simplelist problem after converting it to Novdoc"
1. Create bigfile and directories
   
   ```
   GITDIR="/local/git"
   cd ${GITDIR}/doc-sle
   rm -rf build/
   mkdir -p build/.tmp
   mkdir -p ${GITDIR}/doc-sle/build/SLES-deployment/online-docs/
   daps -d DC-SLES-deployment profile
   xmllint --xinclude \
   --output ${GITDIR}/doc-sle/build/.tmp/book.sle.deployment_for_conversion.xml \
   ${GITDIR}/doc-sle/build/.profiled/x86_64_zseries_power_sles/MAIN.SLEDS.xml
   ```
2. Create DB4 files with `daps-xslt/migrate/db5to4/db5to4-withdoctype.xsl` and with `daps-xslt/migrate/db5to4/db5to4.xsl`
   
   ```
   ${GITDIR}/daps/libexec/daps-xslt \
   --output ${GITDIR}/doc-sle/build/.tmp/db4_with_header.xml \
   --stylesheet ${GITDIR}/daps/daps-xslt/migrate/db5to4/db5to4-withdoctype.xsl \
   --file ${GITDIR}/doc-sle/build/.tmp/book.sle.deployment_for_conversion.xml \
   /usr/bin/xsltproc 
   ${GITDIR}/daps/libexec/daps-xslt \
   --output ${GITDIR}/doc-sle/build/.tmp/db4_without_header.xml \
   --stylesheet ${GITDIR}/daps/daps-xslt/migrate/db5to4/db5to4.xsl \
   --file ${GITDIR}/doc-sle/build/.tmp/book.sle.deployment_for_conversion.xml \
   /usr/bin/xsltproc 
   ```
3. Convert `db4_with_header.xml` and `db4_without_header.xml` to NovDoc
   
   ```
   ${GITDIR}/daps/libexec/daps-xslt \
   --output ${GITDIR}/doc-sle/build/SLES-deployment/online-docs/novdoc_from_with_header.xml \
   --stylesheet ${GITDIR}/daps/daps-xslt/migrate/db5to4/db4tonovdoc.xsl \
   --file ${GITDIR}/doc-sle/build/.tmp/db4_with_header.xml \
   /usr/bin/xsltproc  
   ${GITDIR}/daps/libexec/daps-xslt \
   --output ${GITDIR}/doc-sle/build/SLES-deployment/online-docs/novdoc_from_without_header.xml \
   --stylesheet ${GITDIR}/daps/daps-xslt/migrate/db5to4/db4tonovdoc.xsl \
   --file ${GITDIR}/doc-sle/build/.tmp/db4_without_header.xml \
   /usr/bin/xsltproc
   ```
4.  Validate `novdoc_from_with_header.xml` and `novdoc_from_without_header.xml`
   
   ```
   xmllint  --noent --postvalid --noout --xinclude \
   ${GITDIR}/doc-sle/build/SLES-deployment/online-docs/novdoc_from_with_header.xml
   xmllint  --noent --postvalid --noout --xinclude \
   ${GITDIR}/doc-sle/build/SLES-deployment/online-docs/novdoc_from_without_header.xml
   ```

**Comment by tomschr** [comment]
> Nr 5. <entry> elements in a table that contain a remark cause errors:
> - the `remark`-element is deleted

This is done on purpose. A `remark` isn't really useful for online-docs anymore. Additionally, removing it makes it easier and a lot of errors aren't possible anymore.

> - the `para` element is wrapped with another `para`: <para><para>...</para></para>

This should be already fixed in commit 1c891ff5afe0744009ed2780bd3cb755762e930e

**Comment by tomschr** [comment]
> nr. 4 db4->novdoc conversion only works when using `db5to4.xsl` for the db5->db4 conversion first. When using `db5to4-withdoctype.xsl` for the db5->db4 conversion, it produces invalid NovDoc XML

After some investigations, there is one difference: the DTD. In the case of a document with a doctype, the XSLT processor loads the DTD which provides default attributes.
